activerecord/lib/active_record/connection_adapters/postgresql/oid/uuid.rb

```diff
@@ -15,18 +15,23 @@ def type
 
           def changed?(old_value, new_value, _new_value_before_type_cast)
             old_value.class != new_value.class ||
-              new_value && old_value.casecmp(new_value) != 0
+              new_value != old_value
           end
 
           def changed_in_place?(raw_old_value, new_value)
             raw_old_value.class != new_value.class ||
-              new_value && raw_old_value.casecmp(new_value) != 0
+              new_value != raw_old_value
           end
 
           private
             def cast_value(value)
-              casted = value.to_s
-              casted if casted.match?(ACCEPTABLE_UUID)
+              value = value.to_s
+              format_uuid(value) if value.match?(ACCEPTABLE_UUID)
+            end
+
+            def format_uuid(uuid)
+              uuid = uuid.delete("{}-").downcase
+              "#{uuid[..7]}-#{uuid[8..11]}-#{uuid[12..15]}-#{uuid[16..19]}-#{uuid[20..]}"
             end
```

Comment:
This creates a few unnecessary strings if we already have a string in the correct format. Not sure if it's worth optimising to avoid them? 🤔

Reply:
I think it would be worth having a regexp for the canonical representation and save doing the reformatting if it matches.

We can assume that in the common case no reformatting is needed.

Reply:
Added.